botocore/handlers.py

```diff
@@ -265,9 +266,9 @@ def _register_for_operations(config, session, service_name):
                          handler, unique_id=unique_id)
 
 
-def signature_overrides(service_data, service_name, session, **kwargs):
-    scoped_config = session.get_scoped_config()
-    service_config = scoped_config.get(service_name)
+def signature_overrides(event_name, config, signature_version, **kwargs):
+    service_name = event_name.split('.')[1]
```

Comment:
I am not sure if you can rely on parsing ``event_name``.  The service name used in this event comes from the model, and before it came from whatever you specified from ``get_service()`` call from the session. So, for like ``elb`` the event emitted will most likely be ``elasticloadbalancing`` because that is how it is defined in the model. I have not confirmed that is the case. I would need to check it.